cli/command/container/client_test.go

```diff
@@ -1,19 +1,155 @@
 package container
 
 import (
+	"io"
+	"time"
+
 	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/api/types/network"
 	"github.com/docker/docker/client"
 	"golang.org/x/net/context"
 )
 
 type fakeClient struct {
 	client.Client
-	containerInspectFunc func(string) (types.ContainerJSON, error)
+	inspectFunc     func(string) (types.ContainerJSON, error)
+	execInspectFunc func(ctx context.Context, execID string) (types.ContainerExecInspect, error)
 }
 
 func (cli *fakeClient) ContainerInspect(_ context.Context, containerID string) (types.ContainerJSON, error) {
-	if cli.containerInspectFunc != nil {
-		return cli.containerInspectFunc(containerID)
+	if cli.inspectFunc != nil {
+		return cli.inspectFunc(containerID)
 	}
 	return types.ContainerJSON{}, nil
 }
+
+func (cli *fakeClient) ContainerAttach(ctx context.Context, container string, options types.ContainerAttachOptions) (types.HijackedResponse, error) {
```

Comment:
@dnephin why redefining all of thoses ? `fakeClient` is already a `ContainerClient` as it embedded `client.Client` — and even better, if a call to an unexpected function is done, it will fail the test.. By redifining thoses, this is not the case anymore 😓

Reply:
I think eventually we will add all of these anyway because to test every command we will use every client method. I just added them optimistically instead of waiting. If you would prefer to wait until they are used I can remove them.